src/vulkan/pipeline.cc

```diff
@@ -185,10 +221,35 @@ Result Pipeline::CreateDescriptorSets() {
 }
 
 Result Pipeline::CreatePipelineLayout() {
+  // Put all elements of |descriptor_set_layouts_| and
+  // |empty_descriptor_set_layouts_| together. Note that their
+  // order and the order of actually used descriptor sets must
+  // exactly match.
+  std::vector<VkDescriptorSetLayout> layouts;
+  uint32_t desc = 0;
+  size_t empty_descriptor_set_layouts_index = 0;
+  for (size_t i = 0; i < descriptor_set_numbers_.size(); ++i) {
```

Comment:
Instead of storing to lists, why not just store a vector that you allocate to the max descriptor set value, then fill in the missing ones with empty descriptor sets?

Reply:
Updated. I put all together into `descriptor_set_layouts_`.